doc/links.md

```diff
@@ -136,6 +136,40 @@ and have that compiled into the library. This is necessary if you want to use
 the python bindings, since they rely on dynamically loading the datamodel
 libraries.
 
+## The `LinkNavigator` utility
+
+`podio::LinkCollection`s store each link separately even if a given object is
+present in several links. Additionally, they don't offer any real way to look up
+objects that are linked easily (apart from manually looping and comparing
+elements). To alleviate these issues, we provide the `podio::LinkNavigator`
+utility class that facilitates navigating links and lookups. It can be
+constructed from any `podio::LinkCollection` and can then be used to retrieve
+linked objects. E.g.
+
+```cpp
+const auto& recoMcLinks = event.get<edm4hep::RecoMCParticleLinkCollection>("mcrecolinks");
+const auto linkNavigator = podio::LinkNavigator(recoMcLinks);
+
+// For podio::LinkCollections with disparate types just use getLinked
+const auto linkedRecs = linkNavigator.getLinked(mcParticle);
+// If you want to make a point about the direction use getLinked{From,To}
+// This is also necessary if the From and To type in the link collection are the same
+const auto linkedMCs = linkNavigator.getLinkedFrom(recoParticle);
```

Comment:
Yes, exactly. I started the naming for these from the arguments. Maybe we can drop that detail from the documentation as well, as for the majority of links, I assume the templated version will do correct thing in any case. 

Alternatively, we could also introduce a two argument version that takes the direction as some enum?

Reply:
I think a two argument version of `getLinked` with an enum sounds less confusing and more explicit than `getLinkedFrom`.

Reply:
I proposed some word changes to avoid re-using to and from in the documentation strings.

I think instead of `Lookup` `Return` might be more explicit.

`getLinked(recoParticle, ReturnFrom);`

makes it more obvious that we return `from` objects?